src/compiler/watch.ts

```diff
@@ -739,9 +739,9 @@ export function createWatchFactory<Y = undefined>(host: WatchFactoryHost & { tra
 export function createCompilerHostFromProgramHost(host: ProgramHost<any>, getCompilerOptions: () => CompilerOptions, directoryStructureHost: DirectoryStructureHost = host): CompilerHost {
     const useCaseSensitiveFileNames = host.useCaseSensitiveFileNames();
     const hostGetNewLine = memoize(() => host.getNewLine());
-    return {
+    const compilerHost: CompilerHost = {
         getSourceFile: createGetSourceFile(
-            (fileName, encoding) => host.readFile(fileName, encoding),
+            (fileName, encoding) => !encoding ? compilerHost.readFile(fileName) : host.readFile(fileName, encoding),

```

Comment:
Odd to me that this looks like this, given `compilerHost` just defers to `host.readFile` too. Is it expected that `CompilerHost` is missing this parameter?

Reply:
I think we are suppose to deprecate the `encoding = charSet from compilerOptions` Our APIs dont have that consistent everywhere.

Reply:
compilerHost.readFile is normally used for module resolution only and it doesnt support passing encoding...